src/ccs/mod.rs

```diff
@@ -41,15 +41,15 @@ pub struct CCS<C: CurveGroup> {
 impl<C: CurveGroup> CCS<C> {
     /// check that a CCS structure is satisfied by a z vector. Only for testing.
     pub fn check_relation(&self, z: &[C::ScalarField]) -> Result<(), Error> {
-        let mut result = vec![C::ScalarField::zero(); self.m];
+        let mut result = vec![C::ScalarField::zero(); self.n];
 
         for i in 0..self.q {
             // extract the needed M_j matrices out of S_i
             let vec_M_j: Vec<&SparseMatrix<C::ScalarField>> =
                 self.S[i].iter().map(|j| &self.M[*j]).collect();
 
             // complete the hadamard chain
-            let mut hadamard_result = vec![C::ScalarField::one(); self.m];
+            let mut hadamard_result = vec![C::ScalarField::one(); self.n];
```

Comment:
ditto

Reply:
Right, same as in https://github.com/privacy-scaling-explorations/folding-schemes/pull/11#discussion_r1313780647